core/server/master/src/test/java/alluxio/master/meta/UpdateCheckTest.java

```diff
@@ -46,86 +53,135 @@ public void before() {
+    Assert.assertTrue(!UpdateCheck.getFeatureString().contains("embedded"));
+    ServerConfiguration.set(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.EMBEDDED);
+    Assert.assertTrue(UpdateCheck.getFeatureString().contains("embedded"));
+
+    // Rocks
+    ServerConfiguration.set(PropertyKey.MASTER_METASTORE, MetastoreType.ROCKS);
+    Assert.assertTrue(UpdateCheck.getFeatureString().contains("rocks"));
+    ServerConfiguration.set(PropertyKey.MASTER_METASTORE, MetastoreType.HEAP);
+    Assert.assertTrue(!UpdateCheck.getFeatureString().contains("rocks"));
+
+    // Zookeeper
+    ServerConfiguration.set(PropertyKey.ZOOKEEPER_ENABLED, true);
+    Assert.assertTrue(UpdateCheck.getFeatureString().contains("zk"));
+    ServerConfiguration.set(PropertyKey.ZOOKEEPER_ENABLED, false);
+    Assert.assertTrue(!UpdateCheck.getFeatureString().contains("zk"));
+
+    // Backup delegation
+    ServerConfiguration.set(PropertyKey.MASTER_BACKUP_DELEGATION_ENABLED, true);
+    Assert.assertTrue(UpdateCheck.getFeatureString().contains("backupDelegation"));
+    ServerConfiguration.set(PropertyKey.MASTER_BACKUP_DELEGATION_ENABLED, false);
+    Assert.assertTrue(!UpdateCheck.getFeatureString().contains("backupDelegation"));
+
+    // Daily backup
+    ServerConfiguration.set(PropertyKey.MASTER_DAILY_BACKUP_ENABLED, true);
+    Assert.assertTrue(UpdateCheck.getFeatureString().contains("dailyBackup"));
+    ServerConfiguration.set(PropertyKey.MASTER_DAILY_BACKUP_ENABLED, false);
+    Assert.assertTrue(!UpdateCheck.getFeatureString().contains("dailyBackup"));
+
+    // Persistence blacklist
+    ServerConfiguration.set(PropertyKey.MASTER_PERSISTENCE_BLACKLIST, ".tmp");
+    Assert.assertTrue(UpdateCheck.getFeatureString().contains("persistBlackList"));
+    ServerConfiguration.unset(PropertyKey.MASTER_PERSISTENCE_BLACKLIST);
+    Assert.assertTrue(!UpdateCheck.getFeatureString().contains("persistBlackList"));
+
+    // Unsafe direct persist
+    ServerConfiguration.set(PropertyKey.MASTER_UNSAFE_DIRECT_PERSIST_OBJECT_ENABLED, true);
+    Assert.assertTrue(UpdateCheck.getFeatureString().contains("unsafePersist"));
+    ServerConfiguration.set(PropertyKey.MASTER_UNSAFE_DIRECT_PERSIST_OBJECT_ENABLED, false);
+    Assert.assertTrue(!UpdateCheck.getFeatureString().contains("unsafePersist"));
+
+    // Master audit logging
+    ServerConfiguration.set(PropertyKey.MASTER_AUDIT_LOGGING_ENABLED, true);
+    Assert.assertTrue(UpdateCheck.getFeatureString().contains("masterAuditLog"));
+    ServerConfiguration.set(PropertyKey.MASTER_AUDIT_LOGGING_ENABLED, false);
+    Assert.assertTrue(!UpdateCheck.getFeatureString().contains("masterAuditLog"));
+  }
+
+  @Test
+  public void userAgent() throws Exception {
     String userAgentString = UpdateCheck.getUserAgentString("cluster1");
-    Assert.assertTrue(userAgentString.equals(
-        String.format("Alluxio/%s (cluster1; ProductCode:random123code; emr; ec2)",
```

Comment:
is this not how the string ends up being anymore? i prefer the strict equals check than a regex check to be able to potentially catch misformatted entries